### ISSUE TITLE
Several small suggestions for examples\controlnet\train_controlnet.py

### ISSUE THREAD

**Issue by geroldmeisinger** [huggingface/diffusers#5167]
I know that quality-of-life improvements are low-priority for an "example" but I want to compile list of suggestions anyway in case someone wants to tackle it.

**Handle missing resources**

Some steps are pretty heavy (loading SD, loading CN) only to fail because of a missing resource down the line, which may not even be important (like a validation image).

* Check required resources in arguments before running heavy steps
* Print warning if unimportant resources are not available
* Don't stop if unimportant resources are unavailable

**Handle low-disk space**

Running out of disk space happened several times during training and it's especially annoying if you start an over-night run only to find it stopped after 2 checkpoints (yes, you can laugh).

* Print warning if available disk space < 5 * checkpoint size (~2GB)
* Print warning when disk space is likely to run out after every checkpoint (checkpointing_steps * 2GB => "We will run out of disk space in x checkpoints!")
* Handle out of disk space situations: checkpoints before resuming have to be kept at all costs, but new checkpoints should be "deleted by oldest first" to keep training running at all costs.
* Don't save partial checkpoints
* Add a keyboard shortcut to save current checkpoint at any time

**Documentation**

* Create python environment for install requirements
* Explain batch size and gradient accumulation steps
* Explain consequences of memory-saving parameters ("do they change training quality?" "do they came at expense of speed?")
* Increase gradient acculumation steps in default settings (I trained 5 experiments for nothing only the find the effective batch size was much too low, see here https://github.com/lllyasviel/ControlNet/discussions/318#discussioncomment-7176692)
* Provide settings which don't require a HF account

**Misc**

* Handle out-of-memory situation like when the user runs inference during training. I never ran into OOM but I have a suspicion that my trainings failed after swaps?
* print a warning when swapping into shared memory or - god forbid - harddisk
* Extract duplicate code from other train_controlnet_xyz.py scripts and generalize it
* Is seed=0 a problem? https://pytorch.org/docs/stable/generated/torch.Generator.html#torch.Generator.manual_seed . If so, print a warning when using low entropy seeds.
* [Documentation](https://huggingface.co/docs/diffusers/main/en/api/models/controlnet) differs from [readme.md](https://github.com/huggingface/diffusers/tree/main/examples/controlnet#readme) in source (where is the paragraph about Flax?)
* guessmode=true option for validation images
* Related to #5069 and #5070
* add argument to play a sound if an error happened, training finished, or a checkpoint was reached (before or after validation)
* training is not reproducable, at least sha256sums differ at the same checkpoint (with seed=x, and xformers off)

@patrickvonplaten @williamberman @patil-suraj

**Comment by geroldmeisinger** [comment]
another over-night run failed because internet had disconnected... even though everything was available locally 🤦‍♂️

error log:
[ConnectionError.txt](https://github.com/huggingface/diffusers/files/12799430/ConnectionError.txt)
